### PR TITLE
Add configuration option for a custom template directory

### PR DESCRIPTION
**Pull request by FelixScherzinger**
Currently the location for template files is fixed and cannot be altered to personal needs. With this PR template locations can be customized in a way like changing the opanpi file location via input-base-dir is already implemented

### CODE REVIEW

File: client/deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/codegen/OpenApiGeneratorCodeGenBase.java [hunk 1]
```diff
@@ -315,9 +318,17 @@ private Optional<String> getModelNamePrefix(final Config config, final Path open
     }
 
     private Optional<String> getInputBaseDirRelativeToModule(final Path sourceDir, final Config config) {
-        return config.getOptionalValue(getGlobalConfigName(INPUT_BASE_DIR), String.class).map(inputBaseDir -> {
+        return getBaseDirRelativeToModule(config, getGlobalConfigName(INPUT_BASE_DIR), sourceDir);
+    }
+
+    private Optional<String> getTemplateBaseDirRelativeToModule(final Path sourceDir, final Config config) {
```
Comment (ricardozanini) [inline client/deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/codegen/OpenApiGeneratorCodeGenBase.java:324]: The template base directory must be relative to `src/main/resources`.

Comment (ricardozanini) [reply client/deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/codegen/OpenApiGeneratorCodeGenBase.java:324]: IIRC, `baseDir` is not meant to be used by the templates but by the open API folder on the `src/main`.

Comment (FelixScherzinger) [reply client/deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/codegen/OpenApiGeneratorCodeGenBase.java:324]: I just generalized the method for getting a base dir for a certain config value for simplicity reasons.

Comment (FelixScherzinger) [reply client/deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/codegen/OpenApiGeneratorCodeGenBase.java:324]: I am sorry, I guess i do not get your point 😓 
Isn't it unnecessary to check if the templates should be distributed? I thought templates are relevant for the code generation step.
The initial issue, why i created this PR i the first place, was that in my project we use gradle and resources are copied to `build/resources/main/templates` assuming they are stored in `src/main/resources/templates`. So i had to create a custom task copying the templates to `build/classes/templates` to be able to correctly generate code.

Comment (ricardozanini) [reply client/deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/codegen/OpenApiGeneratorCodeGenBase.java:324]: My point is that the code should fetch the custom templates from `src/main`, not the project's root.

Comment (FelixScherzinger) [reply client/deployment/src/main/java/io/quarkiverse/openapi/generator/deployment/codegen/OpenApiGeneratorCodeGenBase.java]: I just updated the method. The sourceDir is taken and the default input base dir stripped to get the source root dir. Custom locations are now enforced to be under `src/main`.

**Review by ricardozanini — APPROVED** [review]

**Review by hbelmiro — APPROVED** [review]

File: docs/modules/ROOT/pages/includes/server-getting-started.adoc
[the remark this answers is not in the document]
Comment (FelixScherzinger) [reply docs/modules/ROOT/pages/includes/server-getting-started.adoc:1]: This change is only affecting `client`

**Review by fjtirado — APPROVED** [review]

File: docs/modules/ROOT/pages/includes/server-getting-started.adoc
[the remark this answers is not in the document]
Comment (FelixScherzinger) [reply docs/modules/ROOT/pages/includes/server-getting-started.adoc:1]: I removed the doc from server documentation 👍

**Pull request merged**